source/common/http/conn_manager_config.h

```diff
@@ -252,10 +253,17 @@ class ConnectionManagerConfig {
   virtual std::chrono::milliseconds delayedCloseTimeout() const PURE;
 
   /**
-   * @return Router::RouteConfigProvider& the configuration provider used to acquire a route
+   * @return Router::RouteConfigProvider* the configuration provider used to acquire a route
    *         config for each request flow.
    */
-  virtual Router::RouteConfigProvider& routeConfigProvider() PURE;
+  virtual Router::RouteConfigProvider* routeConfigProvider() PURE;
+
+  /**
+   * @return Config::ConfigProvider* the configuration provider used to acquire scoped routing
+   * configuration for each request flow. Pointer ownership is _not_ transferred to the caller of
```

Comment:
The `route_config_provider_` returned by this accessor can now be `nullptr`, since enabling scoped RDS means that an HCM scope RDS RouteConfigProvider is no longer necessary.

Another option would be to use `absl::optional<>` but I felt `unique_ptr` is a better fit due to how the delayed allocation happens via `Router::RouteConfigProvider::create()`.

I added the comment about pointer ownership out of an abundance of caution; given Envoy's style of using `unique_ptr`s for ownership transfers, I wouldn't expect a consumer of this interface to confuse the purpose of this API.